libs/langchain/langchain/tools/gmail/get_message.py

```diff
@@ -33,6 +33,27 @@ def _run(
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> Dict:
         """Run the tool."""
+
+        def decode_payload(payload: bytes, charset: Optional[str] = None) -> str:
+            encodings_to_try = [
+                "utf-8",
+                "latin1",
+                "iso-8859-1",
+                "cp1252",
+            ]  # Add more if needed
+
+            if charset:
+                encodings_to_try.insert(0, charset)
```

Comment:
if charset is any of the utf-8/latin1/... it'll be tried twice if for whatever reason it still fails.